### PR TITLE
[v1] Update `color` and `interpolateColors` return type to allow strings

### PR DESCRIPTION
**Pull request by chj-damon**
## Checklist

- [ ] Included code example that can be used to test this change
- [x] Updated TS types
- [x] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [x] Ensured that CI passes

### CODE REVIEW

File: react-native-reanimated.d.ts [hunk 1]
```diff
@@ -357,7 +357,7 @@ declare module 'react-native-reanimated' {
         inputRange: ReadonlyArray<Adaptable<number>>;
         outputColorRange: (string | number)[];
```
Comment (jakub-gonet) [inline react-native-reanimated.d.ts:358]: Could you also change this type to `ReadonlyArray<Adaptable<number | string>>`, like in [`interpolateNode`](https://github.com/software-mansion/react-native-reanimated/pull/1467/files#diff-3eb03eea3c56460a1bdcef23b6d54e490742744b667bb3ebacc1927b8f7d2b49R57)?

**Review by jakub-gonet — COMMENTED** [review]
LGTM, left one comment.

**Comment by jakub-gonet** [comment]
Also, some tests in the [TS test file](https://github.com/software-mansion/react-native-reanimated/blob/2a6c6139a7ac80b4072bb812fdb0d0cad2fee1bc/react-native-reanimated-tests.tsx) would be appreciated.

File: react-native-reanimated.d.ts (hunk 1 above)
Comment (chj-damon) [reply react-native-reanimated.d.ts]: sure thing.

File: react-native-reanimated-tests.tsx [hunk 2]
```diff
@@ -499,3 +501,20 @@ function interpolateNodeTest() {
...
+  const b = useValue(255);
+  const a = useValue(255);
+  return color(r, g, b, a);
+}
+
+function interpolateColorsTest() {
+  const animationValue = useValue(0);
+  const color = interpolateColors(animationValue, {
+    inputRange: [0, 1],
+    outputColorRange: ['red', 'blue'],
```
Comment (jakub-gonet) [inline react-native-reanimated-tests.tsx:517]: This won't work in the real code, we only return string RGB values in the format of `rgba(r,g,b,a)` (where variables are numbers) on the web.

Comment (chj-damon) [reply react-native-reanimated-tests.tsx:517]: then, what it would be like?

Comment (jakub-gonet) [reply react-native-reanimated-tests.tsx:517]: Or actually, it'll work, we use `processColor` internally. I mixed `outputColorRange` with the return type, my bad.

**Review by jakub-gonet — APPROVED** [review]

**Pull request merged**